src/dodal/devices/watsonmarlow323_pump.py

```diff
@@ -0,0 +1,61 @@
+from enum import Enum
+
+from ophyd_async.core import StandardReadable
+from ophyd_async.epics.signal import epics_signal_rw
+
+
+class WatsonMarlow323PumpEnable(str, Enum):
+    DISABLED = "Disabled"
+    ENABLED = "Enabled"
+
+
+class WatsonMarlow323PumpDirection(str, Enum):
+    CLOCKWISE = "CW"
+    COUNTER_CLOCKWISE = "CCW"
+
+
+class WatsonMarlow323PumpState(str, Enum):
+    STOPPED = "STOP"
+    STARTED = "START"
+
+
+class WatsonMarlow323Pump(StandardReadable):
+    """Watson Marlow 323 Pump device"""
+
+    def __init__(self, prefix: str, name: str = "") -> None:
+        self.enabled = epics_signal_rw(
+            WatsonMarlow323PumpEnable,
+            read_pv=prefix + "DISABLE",
+            write_pv=prefix + "DISABLE",
+        )
+
+        self.direction = epics_signal_rw(
+            WatsonMarlow323PumpDirection,
+            read_pv=prefix + "INFO:DIR",
+            write_pv=prefix + "SET:DIR",
+        )
+        self.state = epics_signal_rw(
+            WatsonMarlow323PumpState,
+            read_pv=prefix + "INFO:RUN",
+            write_pv=prefix + "SET:RUN",
+        )
+        self.speed = epics_signal_rw(
+            float, read_pv=prefix + "INFO:SPD", write_pv=prefix + "SET:SPD"
+        )
+        self.limit_high = epics_signal_rw(
+            float, read_pv=prefix + "SET:SPD.DRVH", write_pv=prefix + "SET:SPD.DRVH"
+        )
+        self.limit_low = epics_signal_rw(
+            float, read_pv=prefix + "SET:SPD.DRVL", write_pv=prefix + "SET:SPD.DRVL"
+        )
+
+        self.set_readable_signals(
+            read=[self.state, self.speed, self.direction],
+            config=[
+                self.enabled,
+                self.limit_high,
+                self.limit_low,
+            ],
+        )
```

Comment:
Nit: This and the above signal creation can be replaced with the new ContextManager on StandardReadable:

```suggestion
        with self.add_children_as_readables():
          self.direction = epics_signal_rw(
            WatsonMarlow323PumpDirection,
            read_pv=prefix + "INFO:DIR",
            write_pv=prefix + "SET:DIR",
          )
          self.state = epics_signal_rw(
            WatsonMarlow323PumpState,
            read_pv=prefix + "INFO:RUN",
            write_pv=prefix + "SET:RUN",
          )
          self.speed = epics_signal_rw(
            float, read_pv=prefix + "INFO:SPD", write_pv=prefix + "SET:SPD"
          )
        with self.add_children_as_readables(ConfigSignal):
          self.enabled = epics_signal_rw(
            WatsonMarlow323PumpEnable,
            read_pv=prefix + "DISABLE")
```

Reply:
I did initially consider the using the the context manager as shown in the tutorial but it wasn't clear to me from the name of HintedSignal what it was and I couldn't find any documentation clarifying this. `read` and `config` made more sense and there were more existing examples using this format.

The next device I create I'll use the context manager.

Reply:
https://github.com/bluesky/ophyd-async/issues/349